### PR TITLE
update docstring/docs for transport using request/response models

### PR DESCRIPTION
**Pull request by toppk**
As of merging this PR: https://github.com/encode/httpx/pull/1840

The example in doc and docstring are stale.  This PR fixes that.

### CODE REVIEW

**Review by tomchristie — APPROVED** [review]
Fantastic, thank you!

**Comment by tomchristie** [comment]
Him - looks like black is failing here on one of the comment lines, but I can't see why. Perhaps it's a non-breaking-space character or something, that needs to be a regular whitespace.

Running `scripts/lint` or just running black directly ought to resolve this.

File: httpx/_transports/base.py [hunk 1]
```diff
@@ -23,63 +23,32 @@ def handle_request(self, request: Request) -> Response:
...
 
-        In order to properly release any network resources, the response stream
-        should *either* be consumed immediately, with a call to `stream.read()`,
-        or else the `handle_request` call should be followed with a try/finally
-        block to ensuring the stream is always closed.
+        In order to properly release any network resources, the response
+        stream should *either* be consumed immediately, with a call to 
+        `response.stream.read()`, or else the `handle_request` call should 
+        be followed with a try/finally block to ensuring the stream is 
+        always closed.
```
Comment (tomchristie) [inline httpx/_transports/base.py:34]:
```suggestion
        In order to properly release any network resources, the response
        stream should *either* be consumed immediately, with a call to 
        `response.stream.read()`, or else the `handle_request` call should 
        be followed with a try/finally block to ensuring the stream is 
        always closed.
```

**Comment by tomchristie** [comment]
Ah - got it - trailing whitespace in the docstring.

**Comment by toppk** [comment]
I really don't know what happened here, it looks like you were able to commit to my fork?  I wasn't expecting to have so many commits here.  in any case, let me know if you want me to concisely flatten this PR into one commit.

**Comment by tomchristie** [comment]
> I really don't know what happened here, it looks like you were able to commit to my fork?

Yup! Maintainers can do that when you submit a pull request. 😅

>  let me know if you want me to concisely flatten this PR into one commit.

Nope, thanks. No need. All our pull requests are "squash and merge".

Screenshot 2022-02-08 at 14 56 28

**Pull request merged**